controller/destination/server_test.go

```diff
@@ -79,6 +79,15 @@ func TestSplitDNSName(t *testing.T) {
 		{"example.com..", []string{}, true},
 		{"..example.com.", []string{}, true},
 		{"foo.example.com", []string{"foo", "example", "com"}, false},
+		{"invalid/character", []string{}, true},
+		{"", []string{}, true},
+		{"ALL-CAPS", []string{"ALL-CAPS"}, false},
+		{"This-dns-label-has-63-characters-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", []string{"This-dns-label-has-63-characters-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"}, false},
+		{"0O0", []string{"0O0"}, false},
+		{"-hi", []string{}, true},
+		{"123", []string{}, true},
+		{"This-dns-label-has-64-character-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", []string{}, true},
```

Comment:
Test cases where there are 63- and 64- character names without any hyphens would also be useful.